orchagent/srv6orch.cpp

```diff
@@ -600,6 +784,328 @@ bool Srv6Orch::deleteMysidEntry(const string my_sid_string)
     return true;
 }
 
+uint32_t Srv6Orch::getAggId(const NextHopGroupKey &nhg)
+{
+    SWSS_LOG_ENTER();
+    static uint32_t g_agg_id = 1;
+    uint32_t agg_id;
+    string agg_id_key = nhg.get_srv6_vpn_key();
+
+    if (srv6_prefix_agg_id_table_.find(agg_id_key) != srv6_prefix_agg_id_table_.end()) {
+        agg_id = srv6_prefix_agg_id_table_[agg_id_key].prefix_agg_id;
+        SWSS_LOG_INFO("Agg id already exist, agg_id_key: %s, agg_id %u", agg_id_key.c_str(), agg_id);
+    } else {
+        while (srv6_prefix_agg_id_set_.find(g_agg_id) != srv6_prefix_agg_id_set_.end()) {
+            SWSS_LOG_INFO("Agg id %d is busy, try next", g_agg_id);
+            g_agg_id++;
+            // restart with 1 if flip
+            if (g_agg_id == 0) {
+                g_agg_id = 1;
+            }
+        }
+        agg_id = g_agg_id;
+        srv6_prefix_agg_id_table_[agg_id_key].prefix_agg_id = g_agg_id;
+        // initual ref_count with 0, will be added in increasePrefixAggIdRefCount() later
+        srv6_prefix_agg_id_table_[agg_id_key].ref_count = 0;
+        srv6_prefix_agg_id_set_.insert(g_agg_id);
+        SWSS_LOG_INFO("Agg id not exist, create agg_id_key: %s, agg_id %u", agg_id_key.c_str(), agg_id);
+    }
+
+    return agg_id;
+}
+
+void Srv6Orch::deleteAggId(const NextHopGroupKey &nhg)
+{
+    SWSS_LOG_ENTER();
+
+    string agg_id_key = nhg.get_srv6_vpn_key();
+    uint32_t agg_id;
+
+    if (srv6_prefix_agg_id_table_.find(agg_id_key) == srv6_prefix_agg_id_table_.end()) {
+        return;
+    }
+
+    agg_id = srv6_prefix_agg_id_table_[agg_id_key].prefix_agg_id;
+    if (srv6_prefix_agg_id_table_[agg_id_key].ref_count == 0) {
+        srv6_prefix_agg_id_table_.erase(agg_id_key);
+        srv6_prefix_agg_id_set_.erase(agg_id);
+        SWSS_LOG_INFO("Delete Agg id %d, agg_id_key: %s", agg_id, agg_id_key.c_str());
+    }
+    else
+    {
+        SWSS_LOG_INFO("Referencing this prefix agg id %u : %u", agg_id, srv6_prefix_agg_id_table_[agg_id_key].ref_count);
+    }
+}
+
+void Srv6Orch::increasePrefixAggIdRefCount(const NextHopGroupKey &nhg)
+{
+    SWSS_LOG_ENTER();
+    string k = nhg.get_srv6_vpn_key();
```

Comment:
When the set of NHs are same, but the order of NHs in NHG key are different, does it have a different key? If so, we will use different prefix_agg_id. Could you please clarify this case?

Reply:
It should be treated as same key, I fixed it.
Appreciated for your review comments~